src/unix/haiku/mod.rs

```diff
@@ -352,6 +352,66 @@ s_no_extra_traits! {
         __unused1: *mut ::c_void, // actually a function pointer
         pub sigev_notify_attributes: *mut ::pthread_attr_t,
     }
+
+    #[cfg(target_arch = "x86_64")]
+    pub struct fpu_state {
+        pub control: ::c_ushort,
+        pub status: ::c_ushort,
+        pub tag: ::c_ushort,
+        pub opcode: ::c_ushort,
+        pub rip: ::c_ulong,
+        pub rdp: ::c_ulong,
+        pub mxcsr: ::c_uint,
+        pub mxcsr_mask: ::c_uint,
+        _fpreg: [::c_uchar; 16],
+        _xmm: [[::c_uchar; 16]; 16],
+        _reserved_416_511: [[::c_uchar; 3]; 32],
+    }
+
+    #[cfg(target_arch = "x86_64")]
+    pub struct xstate_hdr {
+        pub bv: ::c_ulong,
+        pub xcomp_bv: ::c_ulong,
+        _reserved: [[::c_uchar; 2]; 24],
```

Comment:
Again, any reason why you deviate from the header here? it is just defined as 48 c_chars.